x/nexus/keeper/lockable_asset.go

```diff
@@ -31,26 +31,13 @@ type lockableAsset struct {
 
 // newLockableAsset creates a coin struct, assign a coin type and normalize the denom if it's a ICS20 token
 func newLockableAsset(ctx sdk.Context, nexus types.Nexus, ibc types.IBCKeeper, bank types.BankKeeper, coin sdk.Coin) (lockableAsset, error) {
-	denom := coin.GetDenom()
-
-	coinType, err := getCoinType(ctx, nexus, ibc, denom)
+	coinType, normalizedCoin, err := normalizeCoin(ctx, nexus, ibc, coin)
```

Comment:
what's the thought on move to separate function? I feels harder to read vs getCoinType + normalize in the constructor.

Reply:
Because getting the coin type isn't sufficient to normalize. We ended up having to check if it's an IBC coin again instead which defeats the purpose. I also don't want to add two different `ICS20` coin types to distinguish just for the purpose of normalization since it doesn't matter later for lock/mint
`if coinType == types.ICS20 && isIBCDenom(denom) {`

Reply:
I can also move all the logic of `normalizeCoin` into this constructor itself. No need for separate getCoinType/normalize, it's the constructor's responsibility to handle this.